pczt/src/common.rs

```diff
@@ -0,0 +1,202 @@
+use std::collections::BTreeMap;
+
+use getset::Getters;
+use serde::{Deserialize, Serialize};
+
+use crate::roles::combiner::merge_map;
+
+pub(crate) const FLAG_INPUTS_MODIFIABLE: u8 = 0b0000_0001;
+pub(crate) const FLAG_OUTPUTS_MODIFIABLE: u8 = 0b0000_0010;
+pub(crate) const FLAG_HAS_SIGHASH_SINGLE: u8 = 0b0000_0100;
```

Comment:
```suggestion
/// Inputs/spends can be neither added nor removed.
pub(crate) const FLAG_INPUTS_MODIFIABLE: u8 = 0b0000_0001;
/// Outputs can be neither added nor removed.
pub(crate) const FLAG_OUTPUTS_MODIFIABLE: u8 = 0b0000_0010;
/// At least one input was signed using SIGHASH_SINGLE (with or without ANYONECANPAY).
pub(crate) const FLAG_HAS_SIGHASH_SINGLE: u8 = 0b0000_0100;
```

Reply:
> The design of these flags seems to suggest that we should be able to interleave adding/removing inputs and signing

If you're making a transparent-only transaction then in theory you could do so. This aspect is carried over from the existing deployed PSBT design, in order to make adoption of PCZTs easier in theory (by reducing the delta).

> One possibility is to have a separate `FLAG_SHIELDED_MODIFIABLE` that is always set when we sign any input.

I'm on board with this; then the existing two flags become simply about transparent inputs and outputs, which further reduces the semantic delta between PSBTs and PCZTs.